qa/common/trace_summary.py

```diff
@@ -259,6 +261,117 @@ def summarize(frontend, traces):
+    # - parent output
+    print("Data Flow:")
+
+    # Order traces by id to be more intuitive if 'show_trace'
+    traces = sorted(traces, key=lambda t: t.get('id', -1))
+
+    # {3: [4, 5, 6], 4: [7]}
+    dataflow_parent_map = dict()
+    for trace in traces:
+        if "id" not in trace:
+            continue
+        if "parent_id" in trace:
+            if trace["parent_id"] not in dataflow_parent_map:
+                dataflow_parent_map[trace["parent_id"]] = []
+            dataflow_parent_map[trace["parent_id"]].append(trace["id"])
+
+    first_parent_id = list(dataflow_parent_map.items())[0][0]
+
+    # {3: {4: {7: None}, 5: None, 6: None}}
+    dataflow_tree_map = dict()
+    append_dataflow_tensor(dataflow_tree_map,
+                           first_parent_id,
+                           dataflow_parent_map,
+                           traces)
+
+    print_dataflow_tensor(dataflow_tree_map, traces, tab_num=0)
+
+
+def append_dataflow_tensor(dataflow_tensor_map,
+                           parent_id,
+                           dataflow_tree_map,
+                           traces):
+    if parent_id not in dataflow_tree_map:
+        dataflow_tensor_map[parent_id] = None
+        return
+
+    child_tensor_map = dict()
+    dataflow_tensor_map[parent_id] = child_tensor_map
+
+    child_ids = dataflow_tree_map[parent_id]
+    for child_id in child_ids:
+        append_dataflow_tensor(child_tensor_map, child_id,
+                               dataflow_tree_map, traces)
+
+
+def print_dataflow_tensor(dataflow_tree_map, traces, tab_num):
+    for parent_id in dataflow_tree_map:
+        print_tensor_by_id(parent_id, traces, tab_num)
+
+        if dataflow_tree_map[parent_id] is None:
+            continue
+
+        print_dataflow_tensor(dataflow_tree_map[parent_id], traces, tab_num+1)
+
+
+def print_tensor_by_id(id, traces, tab_num):
+    tabs = "\t"*(tab_num+1)
+
+    print("{0}{1}".format(tabs, "*"*50))
+    for trace in traces:
+        # print model name and version
+        if "id" in trace and "model_name" in trace and "model_version" in trace and "timestamps" in trace and trace["id"] == id:
+            print("{0}Name:{1}".format(
+                tabs, trace["model_name"]))
+            print("{0}Version:{1}".format(
+                tabs, trace["model_version"]))
+        # print data
+        if "id" in trace and "activity" in trace:
+            if trace["id"] == id and trace["activity"] == "7":
+                print("{0}{1}:".format(tabs, "INPUT"))
+                print("{0}\t{1}: {2}".format(tabs, trace["tensor"]["name"],
+                      get_numpy_array(trace["tensor"])))
+            elif trace["id"] == id and trace["activity"] == "8":
+                print("{0}{1}:".format(tabs, "OUTPUT"))
+                print("{0}\t{1}: {2}".format(tabs, trace["tensor"]["name"],
+                      get_numpy_array(trace["tensor"])))
+    print("{0}{1}".format(tabs, "*"*50))
+
+
+TRITON_TYPE_TO_NUMPY = {
+    1: bool,
```

Comment:
Wha are these magic numbers. At the least put comments indicating the corresponding enum name